backends/views.py

```diff
@@ -18,12 +18,14 @@ def complete(request, *args, **kwargs):
     # social_core can get confused on which User should get the SocialAuth object.
     if request.user.is_authenticated:
         key = "{}_state".format(request.backend.name)
+        redirect_url = request.session.get("next")
         backend_state = request.session.get(key)
         logout(request)
         # logout will clear the session, this preserves the backend session state. We need to do
         # this so that this workflow will validate correctly (EdxOrgOAuth2.validate_state).
         # key is the same one used in EdxOrgAuth2.get_session_state().
         request.session[key] = backend_state
+        request.session["next"] = redirect_url
```

Comment:
Can you add a test to make sure that these two session keys are being copied over?

Reply:
@rhysyngsun at this point, I am not sure how we can test this, however I have already added a [test](https://github.com/mitodl/micromasters/pull/4253/files#diff-58c973bba17544e492d75329ea514f14R300) for testing `next` parameter is preserved in `request.session` when user is already logged in.

Reply:
I only see a test for the changes to `pipeline_api.py` and not one for these changes. There is an existing test for functionality in this method in `backends/views_test.py` so that should give you an idea of how to test this.